homeassistant/components/gogogate2/sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Support for Gogogate2 temperature."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+from homeassistant.const import (
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import Entity
+
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "gogogate2"
+
+NOTIFICATION_ID = "gogogate2_temperature"
+NOTIFICATION_TITLE = "Gogogate2 Temperature"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Please move the config section under the gogogate2 domain in configuration.yaml by creating a config schema in the gogogate2 component.

Both the cover and sensor platforms need to be updated.

Reply:
Do you have an example of a component that implements it this way?

Reply:
You can take relevant parts from the tuya integration.

Is this a local only device or are we hitting any cloud servers? This question is important for how we should design the polling for updates mechanism.

Reply:
This is a local only device

Reply:
Good! Then we can keep the default automatic polling in the entities, as is done at the moment. Just give each entity access to the api instance when creating the entity.